fpga_interchange/fpga_interchange.cpp

```diff
@@ -248,6 +274,37 @@ static void emit_net(
         }
     }
 }
+static void find_non_synthetic_edges(const Context * ctx, WireId root_wire,
+        const std::unordered_map<WireId, std::vector<PipId>> &pip_downhill,
+        std::vector<PipId> *root_pips) {
+    std::vector<WireId> wires_to_expand;
+
+    wires_to_expand.push_back(root_wire);
+    while(!wires_to_expand.empty()) {
+        WireId wire = wires_to_expand.back();
+        wires_to_expand.pop_back();
+
+        auto downhill_iter = pip_downhill.find(wire);
+        if(downhill_iter == pip_downhill.end()) {
+            if(root_wire != wire) {
+                log_warning("Wire %s never entered the real fabric?\n",
```

Comment:
should this be a warning or an error?

Reply:
This is a warning right now because the constant merging logic always emits the VCC/GND root, even if unused.  Then when walking that root, it won't enter the real fabric.  In the long term we can prune away an unused VCC/GND root and make this an error again.

Reply:
That's fair as a temporary thing; from an end-user point of view I don't much like the idea of a warning that isn't immediately end-user-actionable but I imagine it will be long gone by the time this is end-user-read.